blockchain/chain.go

```diff
@@ -761,9 +772,15 @@ func (b *BlockChain) connectBlock(node *blockNode, block, parent *dcrutil.Block,
 	// Calculate the exact subsidy produced by adding the block.
 	subsidy := CalculateAddedSubsidy(block, parent)
 
+	// Calcultate the next stake difficulty.
+	nextStakeDiff, err := b.CalcNextRequiredStakeDifficulty()
```

Comment:
You should use `b.calcNextRequiredStakeDifficulty(node)` instead to be sure you're calculating it for the correct node.  Otherwise, the tip isn't updated yet, so you'll be off by one.